llvm/lib/CodeGen/CalcSpillWeights.cpp

```diff
@@ -257,7 +257,9 @@ float VirtRegAuxInfo::weightCalcHelper(LiveInterval &LI, SlotIndex *Start,
       return -1.0f;
     }
 
-    float Weight = 1.0f;
+    // Force Weight onto the stack so that x86 doesn't add hidden precision,
+    // similar to HWeight below.
+    volatile float Weight = 1.0f;
```

Comment:
Can we wrap this in a macro in Support somewhere to only do this if using degenerate x87 handling?

Reply:
Sure, maybe a specific `stack_float_t` type?